File: core-arch/src/main/java/com/huyingbao/core/arch/store/RxStoreKey.java
package com.huyingbao.core.arch.store;

import androidx.lifecycle.ViewModel;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import dagger.MapKey;

/**
 * Qualifier注解用来区分同一纬度下两种不同的创建实例的方法，
 * <p>
 * 区分不同的{@link RxStore}子类 对{@link dagger.multibindings.ClassKey}扩展。
 * <p>
 * Created by liujunfeng on 2019/1/1.
 */
@Documented
@Target({ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@MapKey
public @interface RxStoreKey {
    Class<? extends ViewModel> value();
}

File: core-arch/src/main/java/com/huyingbao/core/arch/store/RxStoreKey.kt
package com.huyingbao.core.arch.store

import androidx.lifecycle.ViewModel
import dagger.MapKey
import kotlin.reflect.KClass

/**
 * Qualifier注解用来区分同一纬度下两种不同的创建实例的方法，
 *
 *
 * 区分不同的[RxStore]子类 对[dagger.multibindings.ClassKey]扩展。
 *
 *
 * Created by liujunfeng on 2019/1/1.
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
@MapKey
annotation class RxStoreKey(val value: KClass<out ViewModel>)
